src/Core/Domain/CartRule/ValueObject/CartRuleAction/AmountDiscountAction.php

```diff
@@ -36,7 +39,7 @@
  * It can optionally have free shipping and gift product.
  * It cannot have percentage discount.
  */
-final class AmountDiscountAction implements CartRuleActionInterface
+class AmountDiscountAction implements CartRuleActionInterface
```

Comment:
AmountDiscount action should not be aware about free shipping. This breaks separation of concerns